src/compiler/scala/tools/nsc/typechecker/ContextErrors.scala

```diff
@@ -427,7 +427,7 @@ trait ContextErrors {
 
       //typedFunction
       def MaxFunctionArityError(fun: Tree) = {
-        issueNormalTypeError(fun, "implementation restricts functions to " + definitions.MaxFunctionArity + " parameters")
+        issueNormalTypeError(fun, s"functions may not have more than ${definitions.MaxFunctionArity} parameters")
```

Comment:
If you want to be extra precise, you can pass `numVparams` from `typedFunction` in here.

Reply:
Done!